stacker_blueprints/empire/controller.py

```diff
@@ -0,0 +1,177 @@
+            "type": "AWS::EC2::VPC::Id",
+            "description": "Vpc Id"},
+        "DefaultSG": {
+            "type": "AWS::EC2::SecurityGroup::Id",
+            "description": "Top level security group."},
+        "PrivateSubnets": {
+            "type": "List<AWS::EC2::Subnet::Id>",
+            "description": "Subnets to deploy private instances in."},
+        "AvailabilityZones": {
+            "type": "CommaDelimitedList",
+            "description": "Availability Zones to deploy instances in."},
+        "InstanceType": {
+            "type": "String",
+            "description": "Empire AWS Instance Type",
+            "default": "m3.medium"},
+        "MinHosts": {
+            "type": "Number",
+            "description": "Minimum # of empire minion instances.",
+            "default": "2"},
+        "MaxHosts": {
+            "type": "Number",
+            "description": "Maximum # of empire minion instances.",
+            "default": "3"},
+        "SshKeyName": {
+            "type": "AWS::EC2::KeyPair::KeyName"},
+        "ImageName": {
+            "type": "String",
+            "description": (
+                "The image name to use from the AMIMap (usually found in the"
+                " config file.)"
+            ),
+            "default": "empire"},
+        "EmpireDBSecurityGroup": {
+            "type": "AWS::EC2::SecurityGroup::Id",
+            "description": "Security group of Empire database."},
+        "DockerRegistry": {
+            "type": "String",
+            "description": (
+                "Optional docker registry where private images are located."
+            ),
+            "default": "https://index.docker.io/v1/"},
+        "DockerRegistryUser": {
+            "type": "String",
+            "description": "User for authentication with docker registry."},
+        "DockerRegistryPassword": {
+            "type": "String",
+            "no_echo": True,
+            "description": (
+                "Password for authentication with docker registry."
+            )},
+        "DockerRegistryEmail": {
+            "type": "String",
+            "description": "Email for authentication with docker registry."},
+    }
+
+    def create_security_groups(self):
+        t = self.template
+
+        t.add_resource(
+            ec2.SecurityGroup(
+                CLUSTER_SG_NAME,
+                GroupDescription=CLUSTER_SG_NAME,
+                VpcId=Ref("VpcId")))
+
+        t.add_output(
+            Output("SecurityGroup", Value=Ref(CLUSTER_SG_NAME)))
+
+        # Allow access to the DB
+        t.add_resource(
+            ec2.SecurityGroupIngress(
+                "EmpireControllerDBAccess",
+                IpProtocol="tcp", FromPort=5432, ToPort=5432,
+                SourceSecurityGroupId=Ref(CLUSTER_SG_NAME),
+                GroupId=Ref("EmpireDBSecurityGroup")))
+
+    def create_ecs_cluster(self):
+        t = self.template
+        t.add_resource(ecs.Cluster("EmpireControllerCluster"))
```

Comment:
Any idea what the naming of this cluster ends up looking like?  I'm actually a little leery of putting the Cluster in here - it's kind of nice to have it be a separate entity.  Even if the Controllers get broken by a cloudformation update, at least you know the old cluster is still ok.  Maybe it's not that big a deal though.